tools/db_crashtest.py

```diff
@@ -594,7 +574,30 @@ def finalize_and_sanitize(src_params):
     if dest_params.get("create_timestamped_snapshot_one_in", 0) > 0:
         dest_params["txn_write_policy"] = 0
         dest_params["unordered_write"] = 0
-
+    # WAL data can be lost when `manual_wal_flush_one_in > 0` and `FlushWAL()`` is not
+    # explictly called by users of RocksDB (in our case, db stress).
+    # Therefore the recovery from such potential WAL data loss is a prefix recovery
+    # that requires `sync_fault_injection=1` to verify.
+    if dest_params.get("sync_fault_injection", 0) == 0:
+        dest_params["manual_wal_flush_one_in"] = 0
+    if dest_params.get("manual_wal_flush_one_in", 0) > 0:
+        dest_params["sync_fault_injection"] = 1
+    if (
```

Comment:
I moved this existing processing down to avoid splitting 
```
if dest_params.get("sync_fault_injection", 0) == 0:
        dest_params["manual_wal_flush_one_in"] = 0
    if dest_params.get("manual_wal_flush_one_in", 0) > 0:
        dest_params["sync_fault_injection"] = 1
```
Both of these two are needed for normal stress run where `sync_fault_injection` and `manual_wal_flush_one_in` can each be 0/1.

Moving this existing processing down is also proved to be safe as (1) all writes of `disable_wal` and `sync_fault_injection` is now above this (2) there are only writes `dest_params["ingest_external_file_one_in"] = 0` (3) there are only writes `dest_params["enable_compaction_filter"] = 0` and one read `if dest_params.get("enable_compaction_filter", 0) == 1:`